File: platform/how-to-get-started/4.database/check_elasticsearch.mjs
#!/usr/bin/env node
/** Validate Elasticsearch connection status. */

import { ElasticsearchConfig, checkConnection } from "@internal/db-connection-elasticsearch";

const start = performance.now();
try {
  const config = new ElasticsearchConfig();
  const configInfo = {
    host: config.options.host,
    port: config.options.port,
    scheme: config.options.scheme,
    vendor: config.options.vendorType,
    useTls: config.options.useTls,
    verifyCerts: config.options.verifyCerts,
    sslShowWarn: config.options.sslShowWarn,
    hasCaCerts: !!config.options.caCerts,
  };
  console.log("[elasticsearch] checking connection", JSON.stringify(configInfo));
  const result = await checkConnection(config);
  const latency_ms = Math.round(performance.now() - start);
  if (result.success) {
    console.log(JSON.stringify({
      status: "ok",
      service: "elasticsearch",
      latency_ms,
      ...configInfo,
      info: result.info,
    }, null, 2));
  } else {
    const error = {
      name: "ConnectionError",
      message: result.error || "Connection check returned unsuccessful",
      code: null,
      cause: null,
    };
    console.error("[elasticsearch] connection failed", JSON.stringify({ latency_ms, error }));
    console.log(JSON.stringify({
      status: "error",
      service: "elasticsearch",
      latency_ms,
      ...configInfo,
      error,
    }, null, 2));
    process.exitCode = 1;
  }
} catch (err) {
  const latency_ms = Math.round(performance.now() - start);
  const error = {
    name: err.name || "Error",
    message: err.message,
    code: err.code || null,
    cause: err.cause
      ? { name: err.cause.name, message: err.cause.message, code: err.cause.code || null }
      : null,
  };
  console.error("[elasticsearch] connection failed", JSON.stringify({ latency_ms, error }));
  console.log(JSON.stringify({
    status: "error",
    service: "elasticsearch",
    latency_ms,
    error,
  }, null, 2));
  process.exitCode = 1;
}

File: platform/how-to-get-started/4.database/check_elasticsearch.py
#!/usr/bin/env python3
"""Validate Elasticsearch connection status."""

import asyncio
import json
import sys
import time

from db_connection_elasticsearch import ElasticsearchConfig, check_connection


async def main():
    start = time.monotonic()
    config = ElasticsearchConfig()
    config_info = {
        "host": config.host,
        "port": config.port,
        "scheme": config.scheme,
        "vendor": config.vendor_type,
        "use_tls": config.use_tls,
        "verify_certs": config.verify_certs,
        "ssl_show_warn": config.ssl_show_warn,
        "has_ca_certs": bool(config.ca_certs),
    }
    print(f"[elasticsearch] checking connection {json.dumps(config_info)}", file=sys.stderr)
    try:
        result = await check_connection(config)
        latency_ms = round((time.monotonic() - start) * 1000)
        if result.get("success"):
            output = {
                "status": "ok",
                "service": "elasticsearch",
                "latency_ms": latency_ms,
                **config_info,
                "info": result.get("info"),
            }
        else:
            error = {
                "name": "ConnectionError",
                "message": result.get("error", "Connection check returned unsuccessful"),
                "code": None,
                "cause": None,
            }
            print(f"[elasticsearch] connection failed ({latency_ms}ms) — {error['message']}", file=sys.stderr)
            output = {
                "status": "error",
                "service": "elasticsearch",
                "latency_ms": latency_ms,
                **config_info,
                "error": error,
            }
    except Exception as exc:
        latency_ms = round((time.monotonic() - start) * 1000)
        error = {
            "name": type(exc).__name__,
            "message": str(exc),
            "code": getattr(exc, "errno", None) or getattr(exc, "code", None),
            "cause": (
                {
                    "name": type(exc.__cause__).__name__,
                    "message": str(exc.__cause__),
                    "code": getattr(exc.__cause__, "code", None),
                }
                if exc.__cause__
                else None
            ),
        }
        print(f"[elasticsearch] connection failed ({latency_ms}ms) — {error['message']}", file=sys.stderr)
        output = {
            "status": "error",
            "service": "elasticsearch",
            "latency_ms": latency_ms,
            **config_info,
            "error": error,
        }

    print(json.dumps(output, indent=2))
    sys.exit(0 if output["status"] == "ok" else 1)


if __name__ == "__main__":
    asyncio.run(main())
